tests/foreman/ui/test_repository.py

```diff
@@ -864,3 +864,24 @@ def test_positive_delete_random_docker_repo(session, module_org):
         # Check whether others repositories are not touched
         for product_name, repo_name in entities_list:
             assert session.repository.search(product_name, repo_name)[0]['Name'] == repo_name
+
+
+@tier2
+def test_positive_recommended_repos(session, module_org):
+    """list recommended repositories using
+     On/Off 'Recommended Repositories' toggle.
+
+    :id: 1ae197d5-88ba-4bb1-8ecf-4da5013403d7
+
+    :expectedresults: Shows repositories as per On/Off 'Recommended Repositories'.
+
+    :CaseLevel: Integration
+    """
+    manifests.upload_manifest_locked(module_org.id)
+    with session:
+        session.organization.select(module_org.name)
+        rrepos_on = session.redhatrepository.read(recommended_repo='on')
+        assert rrepos_on
```

Comment:
Can we make better assertions than that? "recommended returns some data, and non-recommended returns some data, and second data has more items than first" isn't particularly strong.

In `robottelo/constants.py` we have `REPOS` and `REPOSET` variables. They look similar to what we can see in red hat repositories. Perhaps we can use them and verify that they appear in both recommended and non-recommended? Or one specific of them is in list? Or random one is in lists?

Reply:
>> Can we make better assertions than that? 

It was not added because that does change from version to version .. (some repos are available under recommended repos for version 6.5 and some for 6.6).

It's  added now to keep a data check around it .

>> recommended returns some data, and non-recommended returns some data, and second data has more items than first" isn't particularly strong.

As checked the same over version 6.3/4/5/6 it found same (second data has more items than first) behaviour.

Reply:
Thank you, Vijay